prime-dotnet-webapi/Controllers/EnrolleesController.cs

```diff
@@ -558,5 +558,32 @@ public async Task<ActionResult> sendEnrolleeReminderEmail(int enrolleeId)
 
             return NoContent();
         }
+
+        // POST: api/Enrollees/5/self-declaration-document/3?{GUID}
+        /// <summary>
+        /// Create Self Declaration Document Link
+        /// </summary>
+        /// <param name="enrolleeId"></param>
+        /// <param name="selfDeclarationTypeCode"></param>
+        /// <param name="selfDeclarationDocument"></param>
+        [HttpPost("{enrolleeId}/self-declaration-document/{selfDeclarationTypeCode}", Name = nameof(createSelfDeclarationDocument))]
+        [ProducesResponseType(typeof(ApiBadRequestResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public async Task<ActionResult<SelfDeclarationDocument>> createSelfDeclarationDocument(int enrolleeId, int selfDeclarationTypeCode, SelfDeclarationDocument selfDeclarationDocument)
```

Comment:
This endpoint should probably accept a type code and a document GUID, and construct the document link in the service. At the very least, we shouldnt use the upload date that the frontend sends.